src/timer.rs

```diff
@@ -202,17 +209,10 @@ macro_rules! hal {
                     timer
                 }
 
-                /// Starts listening for an `event`
-                pub fn listen(&mut self, event: Event) {
+                /// Enable or disable the interrupt for the specified [`Event`].
+                pub fn configure_interrupt(&mut self, event: Event, enable: bool) {
```

Comment:
Maybe add a return value, which signals, if the interrupt was enabled or not, because some `Event`s (which are not added yet) can not be used for interrupts.